pkg/apis/v1alpha1/runnable.go

```diff
@@ -61,6 +61,15 @@ type RunnableSpec struct {
 	Selector           *ResourceSelector               `json:"selector,omitempty"`
 	Inputs             map[string]apiextensionsv1.JSON `json:"inputs,omitempty"`
 	ServiceAccountName string                          `json:"serviceAccountName,omitempty"`
+	// +kubebuilder:default={maxFailedRuns: 5, maxSuccessfulRuns: 3}
+	RetentionPolicy RetentionPolicy `json:"retentionPolicy,omitempty"`
```

Comment:
I wonder if this is the type of thing we'd want to define on a per-instance basis, or a per-deployment basis? i.e. - do we want developers to have control of this retention policy for each instance of a runnable they create, or should we have this policy be defined elsewhere and configurable on deployment of cartographer?

Another concern I have is that by defining the defaults here, the spec of the runnable will be modified which may be unexpected.